redhat-appstudio/service-provider-integration-operator
# Simplified token matching

### What does this PR do?
It is changing the way of SPIAccessToken filtering when we are searching it for a SPIAccessTokenBinding. 
This pr makes it identical to 'hostcredential' provider. When tokens are matched by the service provider and that is all. No permission checks are made. 

### Screenshot/screencast of this PR

<img width="1715" alt="Знімок екрана 2022-08-09 о 17 07 31" src="https://user-images.githubusercontent.com/1614429/183686808-4a473865-99be-4218-af14-d8a3d5f94036.png">


### What issues does this PR fix or reference?
- https://issues.redhat.com/browse/SVPI-184

### How to test this PR?
 - deploy `quay.io/skabashn/service-provider-integration-operator:match2_2022_08_11__16_03_45`
 - create binding for read `https://github.com/redhat-appstudio/service-provider-integration-operator`
 - Complete OAuth flow.
 - Make sure token is injected
